src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -236,37 +236,71 @@ export default class PhotoField extends React.Component {
     this.setState({ windowWidth });
   }
 
+  updateMoveButtonsEnabledStates = (canvasData, cropboxData) => {
+    const { height: canvasHeight, width: canvasWidth, left: canvasLeft, top: canvasTop } = canvasData;
+    const { height: cropboxHeight, width: cropboxWidth, left: cropboxLeft, top: cropboxTop } = cropboxData;
+
+    // origin is at center
+    // coodinates are for canvas's container
+    const canvasBoundaries = {
+      minLeft: -canvasWidth + cropboxLeft + cropboxWidth,
+      maxLeft: cropboxLeft,
+      minTop: -canvasHeight + cropboxTop + cropboxHeight,
+      maxTop: cropboxTop
+    };
+
+    const buttonStates = {
+      moveUpDisabled: canvasTop === canvasBoundaries.minTop,
+      moveDownDisabled: canvasTop === canvasBoundaries.maxTop,
+      moveLeftDisabled: canvasLeft === canvasBoundaries.maxLeft,
+      moveRightDisabled: canvasLeft === canvasBoundaries.minLeft
+    };
+    this.setState(buttonStates);
+  }
+
+  cropend = () => {
```

Comment:
I see that this is not camelCase in order to match the expected name, is this kind of thing worthy of a comment or is self-explanatory enough?

Reply:
I'll add a comment about it.